src/detail-page/components/TabJobContent/JobTab.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import styled from 'styled-components';
+import { Button, CircularProgress, Input } from '@material-ui/core';
+import { JobListItem } from './JobListItem';
+import { useFetchJobs } from './useFetchJobs';
+import { Job } from '../../../data/Jobs';
+import hiringPicture from '../../../icons/hiring.svg';
+
+const SearchInput = styled(Input)`
+  width: 250px;
+  height: 35px;
+  border: 1px #CFCFCF solid;
+  margin: ${props => props.theme.spacing(5)}px 0 ${props => props.theme.spacing(1.5)}px ${props => props.theme.spacing(1.5)}px ;
+  padding-left: ${props => props.theme.spacing(1.5)}px 
+`;
+
+const SearchButton = styled(Button)`
+  width: 100px;
+  margin: ${props => props.theme.spacing(1.5)}px;
+  display: block;
+  color: white;
+`;
+
+const HiringImage = styled('object')`
+  width: 250px;
+  float: right;
+  margin-right: ${props => props.theme.spacing(2)}px;
+`;
+
+const SearchSpinner = styled('div')`
+  margin: ${props => props.theme.spacing(5)}px auto;
+  width: 50%;
+`;
+
+const JobList = styled('ul')`
+  padding:0;
+`;
+const JobTab: React.FC = () => {
+  const [role, setRole] = React.useState('');
+  const [city, setCity] = React.useState('');
+  const [url, setUrl] = React.useState('');
+  const handleChangeRole = React.useCallback(e => setRole(e.target.value), []);
+  const handleChangeCity = React.useCallback(e => setCity(e.target.value), []);
+  /* eslint-disable prefer-const */
+  let { jobPosts, isFetching } = useFetchJobs(url);
```

Comment:
Linter complaints often signal that something's not quite ideal in our code :) In this case, it's probably not the best idea to overwrite the value returned from the hook by setting `jobPosts` to `[]` - it creates two sources of truth for the data and that's never good. But we can easily avoid it by resetting the value directly in the hook, the place responsible for managing this data:
```
  useEffect(() => {
    if (!url) {
      return;
    }
    setJobPosts([]);
    const fetchData = async () => {
...
```